llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskCommunicator.java

```diff
@@ -403,9 +401,8 @@ public void indicateError(Throwable t) {
                   + " appId=" + currentQueryIdentifierProto.getApplicationIdString()
                   + " dagId=" + currentQueryIdentifierProto.getDagIndex()
                   + " to node " + node.getHost());
-              if (!processSendError(t)) {
-                callback.setError(null, t);
-              }
+              processSendError(t);
```

Comment:
We have appId till here, why don't we pass this and use this in ``fetchToken`` instead of passing ``null`` there?

Reply:
sorry, I don't get this one, I cannot find the connection between the marked line and appId

Reply:
I meant like that method in the end reaches fetchtoken, and there we are putting appid as null
Can we not propogate the appId from there and use it here
```
  private synchronized void fetchToken() {
    LOG.debug("Trying to fetch a new token...");
    try {
      Token<LlapTokenIdentifier> newToken =
          tokenClient.withCurrentToken(new Token<LlapTokenIdentifier>(token)).getDelegationToken(null); <-- Here, pass appId instead of null
```

Reply:
discussed on slack: I haven't touched getDelegationToken API with this patch, appId is of no relevance (it's not even used later)